clients/java/src/main/java/io/zeebe/client/impl/OAuthCredentialsProviderBuilder.java

```diff
@@ -86,9 +111,16 @@ public OAuthCredentialsProvider build() {
     if (System.getenv("ZEEBE_AUTHORIZATION_SERVER_URL") != null) {
       this.authorizationServerUrl = System.getenv("ZEEBE_AUTHORIZATION_SERVER_URL");
     }
+    if (System.getenv("ZEEBE_CLIENT_CONFIG_PATH") != null) {
+      this.credentialsCachePath = System.getenv("ZEEBE_CLIENT_CONFIG_PATH");
+    }
+  }
 
-    validate();
-    return new OAuthCredentialsProvider(this);
+  private void applyDefaults() {
+    if (credentialsCachePath == null) {
+      this.credentialsCachePath =
+          System.getProperty("user.home") + File.separator + ".camunda/credentials";
```

Comment:
I think you can use `Paths.get(String...)` to let the JVM pick the right separators and all that. Though if we only target Unix then I guess hardcoding `/` is ok? Either way, here we're using a hardcoded `/` and `File.separator`, so we should just decide which (I vote using `Paths.get().toString()` or `File.separator` both times).